R/k-mer-based.R

```diff
@@ -196,14 +198,79 @@ calculate_kmer_enrichment <-
             return(list(
                 dfs = enrichment_dfs,
                 kmers = gsub("T", "U", as.character(names(
-                    background_kmers
-                )))
+                    background_kmers))),
+                kmer_origins = create_kmer_origin_list(background_set, k)
+
             ))
         } else {
             return(NULL)
         }
     }
 
+
+#' @title Create \emph{k}-mer lists with their original sequences
+#'
+#' @description
+#' Counts occurrences of \emph{k}-mers of length \code{k} in the given
+#' set of sequences. A list of the sequences the \emph{k}-mer originated
+#' from is created and returned. If the \emph{k}-mer does not exist in
+#' any of the sequences, the list is empty for that \emph{k}-mer.
+#' 
+#' @param sequences character vector of DNA or RNA sequences that
+#' constitute the whole set of sequences being analyzed
+#' @param k number of characters in \emph{k}-mer
+#'
+#' @return A list with \emph{k}-mers as the identifier and a vector
+#' of sequences for each \emph{k}-mer that the \emph{k}-mer is found in
+#'
+#' @examples # define simple sequence sets for foreground and background
+#' sequence_set <- c(
+#'   "CAACAGCCUUAAUU", "CAGUCAAGACUCC", "CUUUGGGGAAU",
+#'   "UCAUUUUAUUAAA", "AAUUGGUGUCUGGAUACUUCCCUGUACAU",
+#'   "AUCAAAUUA", "AGAU", "GACACUUAAAGAUCCU",
+#'   "UAGCAUUAACUUAAUG", "AUGGA", "GAAGAGUGCUCA",
+#'   "AUAGAC", "AGUUC", "CCAGUAA"
+#' )
+#' kmer_origin_list <- create_kmer_origin_list(sequence_set, k = 6)
+#' 
+#' @importFrom Biostrings DNAStringSet
+#' @importFrom Biostrings RNAStringSet
+#' @importFrom stringr str_trim
+#' @family \emph{k}-mer functions
+#' @export
+create_kmer_origin_list <- 
+  function(sequences, k) {
+    kmer_counts <- data.frame()
+    all_seqs_as_string <- paste(sequences, collapse='')
+    is_RNA <- FALSE
```

Comment:
Should this be a parameter of the function? Right now it's a constant. Or do we need it at all? Should Us always be turned into Ts?

Reply:
The else if statement in line 249 should have contained a line switching is_RNA from FALSE to TRUE, which I forgot to add (it should be added now).

It could be a parameter of the function but I figured it could just be detected based on what is in the code right now so I left it as a variable within.

Reply:
Can you rename it to `is_rna` to fit with conventions?